lesson_19_2/test_app.py

```diff
@@ -0,0 +1,36 @@
+import os
+import requests
+import tempfile
+
+def test_upload_retrieve_image():
+    # Create a temporary directory
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # Create a temporary image file
+        temp_file_path = os.path.join(temp_dir, 'image.jpg')
+        with open(temp_file_path, 'wb') as temp_file:
+            # Upload the image
+            response = requests.post('http://127.0.0.1:8080/upload', files={'image': open(temp_file_path, 'rb')})
+            assert response.status_code == 201
+            image_url = response.json()['image_url']
+
+            # Retrieve the image
+            response = requests.get(f'http://127.0.0.1:8080/image/{image_url.split("/")[-1]}', headers={'Accept': 'image/jpeg'})
```

Comment:
get method should be inside separate test case. It is not so good to mixed up all tests together. Especially when your task was to write 3 test cases - for post, get and delete.

Reply:
Make changes.